### PR TITLE
Support GCS history backend (simpler version)

### PR DESCRIPTION
**Pull request by KengoTODA**
Just like #23, this PR adds a support for GCS history backend.
But it provides less configurations than #23, refer to https://github.com/minamijoyo/tfmigrate-storage/pull/3 for detail.

### CODE REVIEW

File: config/storage_gcs_test.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+}
+`,
+			want: &gcs.Config{
+				Bucket: "tfmigrate-test",
+				Name:   "tfmigrate/history.json",
+			},
+			ok: true,
+		},
+		{
+			desc: "valid (with optional)",
```
Comment (minamijoyo) [inline config/storage_gcs_test.go:37]: This case is the same as valid (required) and there is no optional attributes.

**Comment by minamijoyo** [comment]
Hi @KengoTODA, Thank you for working on this!

I tried to test this branch with docker-compose, but failed to initialize the test fixture for storage_gcs.

```
$ docker-compose build
$ docker-compose run --rm tfmigrate /bin/bash

bash-5.1# mkdir -p tmp/dir1 && cd tmp/dir1
bash-5.1# terraform init -from-module=../../test-fixtures/storage_gcs
Copying configuration from "../../test-fixtures/storage_gcs"...


Initializing the backend...
╷
│ Error: Failed to get existing workspaces: querying Cloud Storage failed: googleapi: Error 401: Anonymous caller does not have storage.objects.list access to the Google Cloud Storage bucket., required
│
│
╵

bash-5.1# terraform -v
Terraform v1.2.8
on linux_amd64
bash-5.1# echo $STORAGE_EMULATOR_HOST
http://fake-gcs-server:4443
```

I guess the error looks related to an issue mentioned in https://github.com/minamijoyo/tfmigrate/pull/23#issuecomment-902352674 by @w1mvy 🤔 

@KengoTODA Was it successful on this branch in your environment?

**Comment by KengoTODA** [comment]
Thanks for your review!  
It's my mistake, I've added necessary code to run the `terraform` and `tfmigrate` in the container.
Sorry for making you got confused!

**Comment by minamijoyo** [comment]
It works now. It's a bit tricky but acceptable to use s3 backend as Terraform state backend for testing gcs storage as tfmigrate history 👍

File: config/storage_gcs_test.go (hunk 1 above)
Comment (minamijoyo) [reply config/storage_gcs_test.go:37]: @KengoTODA It looks the same as the above `valid (required)` case. I'm not sure what is different. Can we remove this case as duplicate?

Comment (KengoTODA) [reply config/storage_gcs_test.go:37]: Yes it should be removed when I removed the `Endpoint` config. I will handle it soon.

**Review by minamijoyo — APPROVED** [review]
I’ve also confirmed that this works with a real gcs bucket. LGTM :+1:

**Comment by KengoTODA** [comment]
Thank you!

**Comment by minamijoyo** [comment]
@KengoTODA I've just released it in v0.3.7 🚀 Thank you for your contribution!

**Pull request merged**